components/core/src/clp/regex_utils/regex_translation_utils.cpp

```diff
@@ -159,6 +171,25 @@ auto dot_state_transition(
     return ErrorCode::Success;
 }
 
+auto escaped_state_transition(
+        TranslatorState& state,
+        string_view::const_iterator& it,
+        string& wildcard_str,
+        [[maybe_unused]] RegexToWildcardTranslatorConfig const& config
+) -> error_code {
+    auto const ch{*it};
+    if (!cRegexEscapeSeqMetaChars.at(ch)) {
+        return ErrorCode::IllegalEscapeSequence;
+    }
+    if (cWildcardMetaChars.at(ch)) {
+        wildcard_str = wildcard_str + cEscapeChar + ch;
+    } else {
+        wildcard_str += ch;
+    }
```

Comment:
```suggestion
    if (cWildcardMetaChars.at(ch)) {
        wildcard_str += cEscapeChar;
    }
    wildcard_str += ch;
```

Reply:
I've thought about doing this, but later when we implement quantifiers, we need to determine the exact token char/string to generate repetitions upon. In that case, it's better to keep a separate if-else structure.